src/Core/Manifest/PrioritySorter.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace SilverStripe\Core\Manifest;
+
+use SilverStripe\Core\Injector\Injectable;
+
+/**
+ * Class PrioritySorter
+ * @package SilverStripe\Core\Manifest
+ */
+class PrioritySorter
+{
+    use Injectable;
+
+    /**
+     * The key that is used to denote all remaining items that have not
+     * been specified in priorities
+     * @var string
+     */
+    protected $restKey = '...rest';
+
+    /**
+     * A map of variables to their values
+     * @var array
+     */
+    protected $variables = [];
+
+    /**
+     * @var array
+     */
+    protected $items;
```

Comment:
A description of the format of these arrays helps with readability :) What is it a map from => to?